internal/mithril-persistence/src/sqlite/connection_pool.rs

```diff
@@ -0,0 +1,85 @@
+use std::{ops::Deref, time::Duration};
+
+use mithril_common::{
+    resource_pool::{Reset, ResourcePool, ResourcePoolItem},
+    StdResult,
+};
+
+use crate::sqlite::SqliteConnection;
+
+/// SqliteConnection wrapper for a pooled connection
+pub struct SqlitePooledConnection(SqliteConnection);
+
+impl SqlitePooledConnection {
+    /// Create a new SqlitePooledConnection
+    pub fn new(connection: SqliteConnection) -> Self {
+        Self(connection)
+    }
+}
+
+impl Deref for SqlitePooledConnection {
+    type Target = SqliteConnection;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Reset for SqlitePooledConnection {}
+
+/// Pool of Sqlite connections
+pub struct SqliteConnectionPool {
+    connection_pool: ResourcePool<SqlitePooledConnection>,
+}
+
+impl SqliteConnectionPool {
+    /// Create a new SqliteConnectionPool
+    pub async fn new(connections: Vec<SqlitePooledConnection>) -> Self {
```

Comment:
Does it need to be asynchronous? Can this constructor be removed since it is not used?